src/utils/checkRole.ts

```diff
@@ -0,0 +1,26 @@
+import httpStatus from 'http-status';
+import APIError from './ApiError';
+
+export function checkUserPatient(req, _res, next) {
+  const { user } = req;
+  if (user.role === 'patient') {
+    next();
+  } else {
+    throw new APIError({
+      message: 'Access Denined',
+      status: httpStatus.FORBIDDEN,
+    });
+  }
+}
+
+export function checkUserDoctor(req, res, next) {
```

Comment:
viết lẫn luộn quá Long ơi, nếu theo function thì function hết luôn, còn viết theo const thì viết theo const hết luôn nha

Reply:
cái này có liên quan gì lẫn lộn nhở, có thể viết function hoặc arrow function vì tính chất hoisting của nó.

Reply:
vậy hoisting trong trường hợp hàm này là gì Long, t k thấy hoisting ở đây

Reply:
việc t thấy ở đây là việc m viết function bthg và arrow function khá lẫn lộn, k có mục đích như m nói

Reply:
ở đây là tránh việc hoisting, tuỳ chỗ mình có thể dùng function hoac arrow function chứ nhỉ

Reply:
cái nào nên dùng phù hợp hay không phù hợp thôi, còn lẫn lộn ở đây là điều gì, từng file chia ra như vậy,nên chú ý tổng quát và dễ đọc của nó chứ m toàn đi vào chi tiết thì tau cũng không biết giải thích sao

Reply:
với lại cái đồ án mình đang rất dễ cháy, còn document nữa, nên là cứ code ra để nó chạy đã, còn mấy cái này thì đánh giá mức tương đối thôi, đánh giá tuyệt đối thì lúc này chưa cần.

Reply:
t hiểu vấn đề là có thể sử dụng cả 2, nhma t thấy m đang sử dụng cả 2 k có mục đích gì cả, cái này là về cách code, phải kĩ tính hơn ở mấy chỗ này mới được. đã k làm còn làm thì làm cho tốt luôn chứ k để những lỗi nhỏ nhỏ như vậy được ( lỗi t nói ở đây kphai là lỗi phía trên nhé ).